### ISSUE TITLE
Editor tools: add link paste to text feature

### ISSUE THREAD

**Issue by jeherve** [scripting/wordlandSupport#26]
### Steps to reproduce

1. Start writing a new post.
2. Copy a URL from another site that you want to include in your post.
3. Highlight the text in your post that should become the anchor text for the link.
4. Paste the URL you had copied before.
    * The URL replaces the selected anchor text.

I would instead expect the selected anchor text to become a link. 

This is standard behavior with more and more online tools nowadays, such as WordPress, Slack, or Notion:

https://github.com/user-attachments/assets/09f9e4b5-fae4-4fde-a2a1-aa80388cca72

It would be nice if WordLand supported that too.

**Comment by scripting** [comment]
@jeherve -- I really like this idea, esp because it's being adopted by other software including WordPress, Slack and Notion. It's exactly these kinds of features I want, where there's a consensus developing, I'm a big believer in [prior art as a design method](http://essaysfromexodus.scripting.com/priorArtDesignMethod) for so many reasons. ;-)

I put in a few hours trying to get this to work, and I'm going to keep trying, but in the meantime I recorded a brief video showing how you can do almost the same thing with the software as it works now.

1. Put the URL on the clipboard.
2. Select the text you want to link from.
3. A toolbar automatically pops up.
4. Click the link icon.
5. Paste the URL.
6. Click the check icon.

https://github.com/user-attachments/assets/4fd1e9ff-9f6f-4a76-82e6-9e1657b29bdb

**Comment by scripting** [comment]
Another reason I love the feature -- it's web-affirming, and the web needs our :heart: these days.

**Comment by jeherve** [comment]
> n the meantime I recorded a brief video showing how you can do almost the same thing with the software as it works now.

Yup, I've been using that pop-up for my links so far, but I've grown so used to just pasting links over selected text that I'll sometimes just paste my link, and thus lose my anchor text:

https://github.com/user-attachments/assets/0be69a25-e2a5-42b2-8047-6a6ac7fe825f

I realize that this is entirely my fault for not using the built-in tooling, but that "Paste to create link" feature is so convenient I got used to it, and miss it a lot when it's not implemented in an editor (it's missing in Google Docs as well, for example).

**Comment by scripting** [comment]
@jeherve -- i really want this feature too, we'll get there. i just spent another few hours trying to get this to work, but this is the most difficult area to work in, the browser and the clipboard -- not exactly well designed. ;-)

**Comment by scripting** [comment]
I posted a question about this in the MediumEditor repo.

https://github.com/yabwe/medium-editor/issues/1609

**Comment by scripting** [comment]
I decided to try this with another AI tool, claude.ai, and immediately saw a problem that didn't for some reason show up last time.

First this is the code they suggested I use.

```
editor.subscribe ('editableKeydown', function (event) {
	if (event.key === 'v' && (event.ctrlKey || event.metaKey)) {
		event.preventDefault ();
		
		navigator.clipboard.readText().then(url => {
			if (url.match(/^https?:\/\//)) {
				const selection = window.getSelection();
				const text = selection.toString();
				
				if (text) {
					const link = `<a href="${url}">${text}</a>`;
					editor.pasteHTML(link, selection);
					}
				}
			});
		}
	});

```
It's getting as far as the navigator.clipboard.readText  call, i don't really know what it does (I probably did 2 weeks ago).

I wish this stuff "just worked." :smile:
